libs/commons-database/src/main/kotlin/gov/cdc/ocio/database/dynamo/DynamoCollection.kt

```diff
@@ -0,0 +1,126 @@
+package gov.cdc.ocio.database.dynamo
+
+import mu.KotlinLogging
+import gov.cdc.ocio.database.persistence.Collection
+import software.amazon.awssdk.enhanced.dynamodb.*
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementRequest
+
+
+/**
+ * DynamoDB collection implementation
+ *
+ * @property enhancedClient DynamoDbEnhancedClient
+ * @property dynamoTableName String
+ * @property tableClassType Class<R>
+ * @property dynamoTable DynamoDbTable<R>?
+ * @constructor
+ */
+class DynamoCollection<R>(
+    private val dynamoDbClient: DynamoDbClient,
+    private val enhancedClient: DynamoDbEnhancedClient,
+    private val dynamoTableName: String,
+    private val tableClassType: Class<R>
+) : Collection {
+
+    private val logger = KotlinLogging.logger {}
+
+    private val dynamoTable = collectionForClass(tableClassType)
+
+    /**
+     * Builds a dynamodb table schema from the provided class type.
+     *
+     * @param classType Class<R>
+     * @return DynamoDbTable<R>?
+     */
+    private fun <R> collectionForClass(classType: Class<R>): DynamoDbTable<R>? {
+        return try {
+            enhancedClient.table(
+                dynamoTableName,
+                TableSchema.fromClass(classType)
+            )
+        } catch (e: Exception) {
+            logger.error("Unable to establish the dynamodb enhanced client for table: ${classType.name} with exception: $e")
+            null
+        }
+    }
+
+    /**
+     * Execute the provided query and return the results as POJOs.
+     *
+     * @param query String?
+     * @param classType Class<T>?
+     * @return List<T>
+     */
+    override fun <T> queryItems(query: String?, classType: Class<T>?): List<T> {
```

Comment:
I like that the interface forces the implementor to provide an empty list rather than null.  We can record this as a possible enhancement, but I kind of like the way it is now.